cdap-app-templates/cdap-etl/cdap-etl-realtime/src/main/java/co/cask/cdap/templates/etl/realtime/ETLRealtimeTemplate.java

```diff
@@ -52,6 +52,8 @@ public void configureAdapter(String adapterName, ETLRealtimeConfig etlConfig, Ad
     super.configureAdapter(adapterName, etlConfig, configurer);
     configurer.addRuntimeArgument(Constants.CONFIG_KEY, GSON.toJson(etlConfig));
     configurer.setInstances(etlConfig.getInstances());
+    // Generate unique id for this adapter creation.
+    configurer.addRuntimeArgument(Constants.Realtime.UNIQUE_ID, String.valueOf(System.currentTimeMillis()));
```

Comment:
Maybe a good idea to have the configurer.addRuntimeArgument to support different primitive types as well. No need to be this PR, but good to note.